src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -748,6 +747,33 @@ public ChildSyntaxList ChildNodesAndTokens()
             return new ChildSyntaxList(this);
         }
 
+        internal ChildSyntaxList.Segment SiblingsAfterSelf()
+        {
+            int initialIndex = 0;
+            int terminalIndex = 0;
+            if (_parent != null)
+            {
+                initialIndex = _parent.GetIndexOfChild(this);
+                Debug.Assert(initialIndex >= 0, "Node couldn't be found in its parent.");
+                initialIndex &= int.MaxValue; // Doesn't affect non-negative numbers.
```

Comment:
i don't understand why this is being done.  we already asserted the value is positive.

Reply:
it is not _expected_ for the index to be negative and a negative index here would be due to some inconsistency,
but this is done to handle such a scenario as gracefully as the baseline code which searches a parent for its child linearly with minimal assumptions

Reply:
> it is not expected for the index to be negative and a negative index here would be due to some inconsistency,

Then this code should be removed.

> but this is done to handle such a scenario as gracefully

Gracefully proceeding when core invariants are broken is an anti-goal of roslyn.  All that leads to are *later* crashes which are even harder to find.

> which searches a parent for its child linearly with minimal assumptions

So a parent shoudl always be able to find its child.  if that fails, trying to somehow make up for it is a *bad* thing.

It would even be fine to crash immediately if this occurs.  so we can get dumps and determine what is going on as quickly as possible.